lib/internal/main/check_syntax.js

```diff
@@ -36,18 +37,29 @@ if (process.argv[1] && process.argv[1] !== '-') {
 
   markBootstrapComplete();
 
-  checkSyntax(source, filename);
+  loadESMIfNeeded(() => checkSyntax(source, filename));
 } else {
   markBootstrapComplete();
 
-  readStdin((code) => {
+  loadESMIfNeeded(() => readStdin((code) => {
     checkSyntax(code, '[stdin]');
-  });
+  }));
 }
 
-async function checkSyntax(source, filename) {
+function loadESMIfNeeded(cb) {
```

Comment:
nit: not sure about the name. It sounds like it would be the primary function here (or check a lot more than it does), but it's a helper.

Reply:
any suggestions?